internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,456 @@
+
+		remoteManifest, err := getRemoteManifest(owner, name, version)
+		if err != nil {
+			return err
+		}
+		pluginManifest = remoteManifest
+	}
+
+	// Select plugin-directory
+	force, err := cmd.Flags().GetBool("force")
+	if err != nil {
+		return err
+	}
+
+	pluginDir, err := getPluginDirFromFlag(cmd)
+	if err != nil {
+		return err
+	}
+
+	var ins *installation
+	if pluginDir == "" {
+		ins, err = getConfluentPlatformInstallation(cmd, force)
+		if err != nil {
+			return err
+		}
+		pluginDir, err = choosePluginDir(ins, force)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Check for, and possibly remove, existing installation
+	if previousInstallations, err := existingPluginInstallation(pluginDir, pluginManifest); err != nil {
+		return err
+	} else if len(previousInstallations) > 0 {
+		output.Println("\nA version of this plugin is already installed and must be removed to continue.")
+		for _, previousInstallation := range previousInstallations {
+			if err := uninstall(previousInstallation, dryRun, force); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Install
+	if err := checkLicenseAcceptance(pluginManifest, force); err != nil {
+		return err
+	}
+
+	output.Printf("\nInstalling %s %s, provided by %s\n", pluginManifest.Title, pluginManifest.Version, pluginManifest.Owner.Name)
+	if dryRun {
+		output.Println("[DRY RUN] Skipping plugin installation.")
+	} else {
+		if utils.DoesPathExist(args[0]) {
+			if err := installFromLocal(pluginManifest, args[0], pluginDir); err != nil {
+				return err
+			}
+		} else {
+			if err := installFromRemote(pluginManifest, pluginDir); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Select and update worker-configs
+	workerConfigs, err := cmd.Flags().GetStringSlice("worker-configs")
+	if err != nil {
+		return err
+	}
+
+	if len(workerConfigs) == 0 {
+		if ins == nil {
+			ins, err = getConfluentPlatformInstallation(cmd, force)
+			if err != nil {
+				return err
+			}
+		}
+		workerConfigs, err = chooseWorkerConfigs(cmd, ins, force)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(workerConfigs) > 0 {
+		updateWorkerMsg := "\nAdding plugin installation directory to the plugin path in the following files:"
+		for _, workerConfig := range workerConfigs {
+			updateWorkerMsg = fmt.Sprintf("%s\n\t* %v", updateWorkerMsg, workerConfig)
+		}
+		output.Println(updateWorkerMsg)
+	}
+
+	for _, workerConfig := range workerConfigs {
+		if err := updateWorkerConfig(pluginDir, workerConfig, dryRun); err != nil {
+			return err
+		}
+	}
+
+	if dryRun {
+		output.Println("\n[DRY RUN] Dry run successfully completed. All requirements are met; you may proceed with the installation.")
+	}
+
+	return nil
+}
+
+func parsePluginId(plugin string) (string, string, string, error) {
+	parsePluginErrorMsg := "plugin not found"
+	parsePluginSuggestions := "Provide either a path to a local file or a plugin id from Confluent Hub with the format: `<owner>/<name>:<version>`."
+
+	ownerNameSplit := strings.Split(plugin, "/")
```

Comment:
You might like this format better:
```suggestion
	x := strings.SplitN(plugin, "/", 2)
	owner, name := x[0], x[1]
```

Reply:
I think I prefer this. There shouldn't be any difference between the function calls themselves when N=2, and I can't assign `x[1]` until I've checked that it exists (which it generally won't with a badly formed id).